src/boosting/boosting.cpp

```diff
@@ -3,13 +3,48 @@
 
 namespace LightGBM {
 
-Boosting* Boosting::CreateBoosting(BoostingType type,
-                         const BoostingConfig* config) {
-  if (type == BoostingType::kGBDT) {
-    return new GBDT(config);
+Boosting* Boosting::CreateBoosting(BoostingType type, const char* filename) {
+  if (filename[0] == '\0') {
+    if (type == BoostingType::kGBDT) {
+      return new GBDT();
+    } else {
+      return nullptr;
+    }
   } else {
-    return nullptr;
+    Boosting* ret = nullptr;
```

Comment:
This part looks like CreateBoosting(const char* filename), can we reuse that part?